### ISSUE TITLE
Support installation from NPM for Nodejs

### ISSUE THREAD

**Issue by AdamMagaluk** [intel-iot-devkit/mraa#44]
Currently installing mraa Nodejs requires installing from `opkg`, this is normal for c++ libraries but rarely in the Nodejs community. In the Nodejs community dependencies are handled through the [npm](https://npmjs.org) package management system, which does support native modules.

I would like help with this anyway possible. I currently have a module [mraa-js](https://github.com/AdamMagaluk/mraa-js) that a) deploys the built mraa.node from opkg b) has a set of mocks for development on non edison boards.

Options for native packaging native modules:
1. Add the .node file to the package, since the module is made to support only a number of systems this approach may be optimal.
2. Add a native build step and let node-gyp build mraa when npm install runs.
3. Others??

Any thoughts would be great.
Adam

**Comment by arfoll** [comment]
@AdamMagaluk would you mind taking down the mraa-js module on npm since I don't think it's required anymore and has caused confusion for some users :/

**Comment by AdamMagaluk** [comment]
Will do. Give a day or two gotta update and test some dependencies that rely on it.

**Comment by jjpr** [comment]
@arfoll, did anything ever come of mraa-mock?  I've found that the inability to install mraa for node on a desktop OS makes it difficult for a developer who has an Edison to collaborate with a developer who doesn't have an Edison.  (It's easier to work on a web UI if you run the server, but the server dies on "require('mraa')".)  For now I've repurposed [the mock from galileo-io](https://github.com/rwaldron/galileo-io/blob/master/test/mraa-mock.js), but my efforts seem like a hack.  If you found a more elegant way, it might benefit others.  Thanks.
